### PR TITLE
Wk 3 tutorial

### PR DESCRIPTION
**Pull request by sunr1**
changed text in servlet message as part of the tutorial

### CODE REVIEW

**Review by tambchop — APPROVED** [review]
Looks good to me =)

Side note: it looks like most of the changes in this pull request are the same as in your last one (Scroll button).. so make sure in the future to create your new branches off of master instead of off of your previous pull request's branch.  Except in cases where your new code is very dependent on your previous code... it looks like in this case, your changes seem to be to different parts of the files than in your "Scroll button" pull request, so it probably could have been good to create this new branch from master.  Let me know if I'm mistaken though!

**Review by tambchop — APPROVED** [review]

**Comment by sunr1** [comment]
@tambchop good point! I was attempting to push to a branch created off master and thought I had so I'll double check next time to make sure that happens!

File: portfolio/pom.xml [hunk 1]
```diff
@@ -35,7 +41,7 @@
         <version>2.2.0</version>
         <configuration>
           <!-- TODO: set project ID. -->
```
Comment (kzweller) [inline portfolio/pom.xml:43]: Remove this TODO now.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -24,9 +24,40 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
```
Comment (kzweller) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:25]: Suggest renaming this something other than "Data" - "Comments" or something, maybe. Up to you

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+
+function scrollUp() {
+    window.scrollTo({top:0, behavior: 'smooth'});
+}
+
+async function getGreeting() {
+  const response = await fetch('/data');
+  const greeting = await response.text();
+  document.getElementById('message-container').innerText = greeting;
```
Comment (kzweller) [inline portfolio/src/main/webapp/script.js:9]: We typically avoid directly manipulating the innerText or innerHTML for security reasons.

Suggest instead appending a textNode:

https://www.w3schools.com/jsref/met_node_appendchild.asp

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -24,9 +24,40 @@
...
+  }
+
+  /**
+   * Converts a ServerStats instance into a JSON string using the Gson library. Note: We first added
+   * the Gson library dependency to pom.xml.
+   */
+  private String convertToJsonUsingGson(Output output) {
+    Gson gson = new Gson();
+    String json = gson.toJson(output);
+    return json;
```
Comment (kzweller) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:61]: You can inline all of this above I believe

String json = new Gson().toJson(output)

And then you dont need this method at all.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -24,9 +24,40 @@
...
+
+  private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("Spongebob");
+    messages.add("Gary");
+    messages.add("Mr. Krabs");
+  }
```
Comment (kzweller) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:36]: As of right now, your messages list is static (i.e. never changing) so there's no point really in having this method. Instead you can just set the declaration of the value as...

private static final List<String> messages = Arrays.asList("Spongebob", "Gary", "Mr. Krabs");

**Review by kzweller — CHANGES_REQUESTED** [review]

**Pull request closed without merging**